### ISSUE TITLE
Preclusion handling in clips

### ISSUE THREAD

**Issue by paradite** [bsmmoon/KBMP#22]

**Comment by paradite** [comment]
@bsmmoon @joey-c 
I have implemented the preclusion separately in clips without modifying the module template:
I just need you guys to assert something like this:

```
(preclusion "CS1010" "CS1010E" "CS1010R" "CS1010J" "CS1101S" "CS1010X")
```

This will be handled by the following rule:

```
; ; Module precluded when preclusion met
(defrule RANK::mark-not-available-preclusion-met
    (module (code ?code1) (status ?status))
    (test(or (eq ?status planned) (eq ?status taken)))
    (preclusion $?preclusionlist)
    (test (member$ ?code1 $?preclusionlist))
    ?precludedmodule <- (module (code ?code2) (status available))
    (test (member$ ?code2 $?preclusionlist))
    =>
    (modify ?precludedmodule (status not-available))
    (printout t "Module " ?code2 " precluded " crlf))
```

Let me know if this is okay for you guys.

**Comment by bsmmoon** [comment]
so these are mutually exclusive and if one of them is taken the rest will be not-available is it? as long as parser can parse the info no problem.

**Comment by bsmmoon** [comment]
Actually currently the parser gives information of preclusion for each module. Will need some extra work to put them together..

**Comment by bsmmoon** [comment]
blocked by #36

**Comment by paradite** [comment]
implemented 00f88df634d9a0475b2c213dc775f784407dfa1b
